### PR TITLE
Add support for non-web process types

### LINKED ISSUE #142: Support non web process types
Today only the web process is started. Suggestion: multi-container pods.

### PR DESCRIPTION
**Pull request by alexguerra-luizalabs**
Following Heroku's tradition of not automatically running dynos or
routing web traffic to non-web processes and independent scaling,
we associate to each app one process type and don't create
kubernetes services when this process is non-web. Fixes #142.

---
This change is [Reviewable](https://reviewable.io/reviews/luizalabs/teresa-api/168)

### CODE REVIEW

File: k8s/apps.go [hunk 1]
```diff
@@ -86,7 +92,25 @@ func (c apps) Create(app *models.App, storage helpers.Storage, tk *Token) error
 	if err := c.k.Deployments().CreateAutoScale(app); err != nil {
 		return err
 	}
-	// creating the loadbalancer
+	// Creating the service
+	if err := c.createOrSkipService(app); err != nil {
+		return err
+	}
+	return nil
```
Comment (drgarcia1986) [inline k8s/apps.go]: Why not?
```go
return c.createOrSkipServevice(app)
```
?

File: models/app_in.go [hunk 2]
```diff
@@ -36,6 +36,12 @@ type AppIn struct {
 	*/
 	Name *string `json:"name"`
 
+	/* app process type
+
+	Pattern: ^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	*/
+	ProcessType *string `json:"processType,omitempty"`
```
Comment (drgarcia1986) [inline models/app_in.go]: Why is it a pointer?

File: k8s/deployments.go [hunk 3]
```diff
@@ -378,7 +378,11 @@ func newWelcomeContainer(app *models.App) (c *api.Container) {
 func newSlugRunnerContainer(app *models.App, slug string, storageType string) (c *api.Container) {
 	// creating runner container
 	c = newContainer(*app.Name, slugRunnerImage)
-	c.Args = []string{"start", "web"}
+	processType := processTypeWeb
+	if app.ProcessType != nil {
+		processType = *app.ProcessType
+	}
+	c.Args = []string{"start", processType}
```
Comment (drgarcia1986) [inline k8s/deployments.go]: Is it really necessary to check this value here?

File: k8s/apps.go (hunk 1 above)
Comment (alexguerra-luizalabs) [inline k8s/apps.go]: Sure, my bad.

File: k8s/deployments.go (hunk 3 above)
Comment (alexguerra-luizalabs) [inline k8s/deployments.go]: It is not, if we edit old apps and put the process type in the annotation this check is useless.
The question is: edit old apps or not? Not sure....I supposed we won't edit but editing is not a big deal either.

File: models/app_in.go (hunk 2 above)
Comment (alexguerra-luizalabs) [inline models/app_in.go]: go-swagger's auto generated code.

File: k8s/deployments.go (hunk 3 above)
Comment (drgarcia1986) [inline k8s/deployments.go]: Good point!
Wen can remove this check after migration and we don't need to dealing with old apps.

Any plans to "fix" this lost information in old apps?

File: models/app_in.go (hunk 2 above)
Comment (drgarcia1986) [inline models/app_in.go]: 😐 🔫

File: k8s/deployments.go (hunk 3 above)
Comment (alexguerra-luizalabs) [inline k8s/deployments.go]: I'm changing my mind to remove this check and force the editing otherwise this check will stay forever.

Comment (drgarcia1986) [inline k8s/deployments.go]: Sounds good.
We can work together to fix it.

**Pull request merged**